src/scripts/main.js

```diff
@@ -1,3 +1,209 @@
 'use strict';
 
-// write code here
+const tbody = document.querySelector('tbody');
+const thead = document.querySelector('thead tr');
+let sortOrder = 'asc';
+let lastSortedColumnIndex = -1;
+
+thead.addEventListener('click', (e) => {
+  const clickedTh = e.target.closest('th');
+
+  const columnIndex = clickedTh.cellIndex;
+  const rows = Array.from(tbody.querySelectorAll('tr'));
+
+  if (columnIndex !== lastSortedColumnIndex) {
+    sortOrder = 'asc';
+  }
+
+  const sortedRows = rows.sort((rowA, rowB) => {
+    const cellA = rowA.querySelectorAll('td')[columnIndex].textContent.trim();
+    const cellB = rowB.querySelectorAll('td')[columnIndex].textContent.trim();
+
+    const cellAHasDigits = hasDigits(cellA);
+    const cellBHasDigits = hasDigits(cellB);
+
+    let comparison = 0;
+
+    if (cellAHasDigits && cellBHasDigits) {
+      comparison = convert(cellA) - convert(cellB);
+    }
+
+    if (!cellAHasDigits && !cellBHasDigits) {
+      comparison = cellA.localeCompare(cellB);
+    }
+
+    return sortOrder === 'asc' ? comparison : -comparison;
+  });
+
+  sortOrder = sortOrder === 'asc' ? 'desc' : 'asc';
+  lastSortedColumnIndex = columnIndex;
+
+  tbody.innerHTML = '';
+
+  sortedRows.forEach((row) => {
+    tbody.appendChild(row);
+  });
+});
+
+tbody.addEventListener('click', (e) => {
+  const clickedRow = e.target.closest('tr');
+
+  if (clickedRow) {
+    tbody.querySelectorAll('tr').forEach((row) => {
+      row.classList.remove('active');
+    });
+
+    clickedRow.classList.add('active');
+  }
+});
+
+document.addEventListener('DOMContentLoaded', () => {
+  const form = document.createElement('form');
+
+  form.classList.add('new-employee-form');
+
+  form.innerHTML = `
+    <label>Name: <input name="name" type="text" data-qa="name" required></label>
+    <label>Position: <input name="position" type="text" data-qa="position" required></label>
+    <label>Office:
+      <select name="office" data-qa="office" required>
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    </select>
+    </label>
+    <label>Age: <input name="age" type="number" data-qa="age" required></label>
+    <label>Salary: <input name="salary" type="number" data-qa="salary" required></label>
+    <button type="submit">Save to table</button>
+  `;
+
+  document.body.appendChild(form);
+
+  form.addEventListener('submit', (e) => {
+    e.preventDefault();
+
+    const eeName = form.elements['name'].value.trim();
+    const position = form.elements['position'].value.trim();
+    const office = form.elements['office'].value;
+    const age = form.elements['age'].value;
```

Comment:
Consider adding a check to ensure that the age input is a valid number before performing the comparison. This will prevent potential errors if a non-numeric value is entered.